[Flaking tests] k8s.io/kubernetes/test/integration/podcertificaterequests.podcertificaterequests

### Which jobs are flaking?

* [sig-release-master-blocking#integration-arm64-master](https://testgrid.k8s.io/sig-release-master-blocking#integration-arm64-master&exclude-non-failed-tests=)

### Which tests are flaking?

* [k8s.io/kubernetes/test/integration/podcertificaterequests.podcertificaterequests](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-arm64-master/2024291118509723648)

### Since when has it been flaking?

* First flaky: Fri, 06 Feb 2026 11:06:02 UTC
* Latest flaky: Sat, 21 Feb 2026 10:41:26 UTC

### Testgrid link

* [Testgrid](https://testgrid.k8s.io/sig-release-master-blocking#integration-arm64-master&exclude-non-failed-tests=)
* [Triage](https://storage.googleapis.com/k8s-triage/index.html?job=ci-kubernetes-integration-arm64-master$&test=k8s.io/kubernetes/test/integration/podcertificaterequests.podcertificaterequests)

### Reason for failure (if possible)

```
	F 2026-02-19 06:43:32 +0530 IST Failed
=== RUN   TestNodeRestriction/node1_can_create_PCR_for_pod_on_n...iting for /healthz to be ok...
--- FAIL: TestNodeRestriction (18.12s)

	F 2026-02-18 02:31:35 +0530 IST Failed
=== RUN   TestNodeRestriction/node1_can_create_PCR_for_pod_on_n...iting for /healthz to be ok...
--- FAIL: TestNodeRestriction (18.21s)

	F 2026-02-17 21:30:37 +0530 IST Failed
=== RUN   TestNodeAuthorization/node1_can_directly_get_pcr1
   ...ting for /healthz to be ok...
--- FAIL: TestNodeAuthorization (3.18s)

	F 2026-02-13 09:53:28 +0530 IST Failed
=== RUN   TestNodeRestriction/node1_can_create_PCR_for_pod_on_n...iting for /healthz to be ok...
--- FAIL: TestNodeRestriction (18.15s)

	F 2026-02-12 22:46:30 +0530 IST Failed
=== RUN   TestNodeRestriction/node1_can_create_PCR_for_pod_on_n...iting for /healthz to be ok...
--- FAIL: TestNodeRestriction (18.11s)


```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig node
/kind flake
cc @kubernetes/release-team-release-signal

--- a/test/integration/podcertificaterequests/podcertificate_integration_test.go
+++ b/test/integration/podcertificaterequests/podcertificate_integration_test.go
@@ -510,7 +510,7 @@ func TestNodeAuthorization(t *testing.T) {
 		// Getting the PCR could fail if there is lag in the node authorizer graph population.
 		// Poll until it succeeds.
 		err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
-			_, err = node1Client.CertificatesV1beta1().PodCertificateRequests("default").Get(ctx, "pcr1", metav1.GetOptions{})
+			_, err := node1Client.CertificatesV1beta1().PodCertificateRequests("default").Get(ctx, "pcr1", metav1.GetOptions{})
 			if err != nil {
 				t.Logf("Get PodCertificateRequest default/pcr1 as node1 failed with error: %v, retrying...", err)
 				return false, nil

--- a/test/integration/podcertificaterequests/podcertificate_integration_test.go
+++ b/test/integration/podcertificaterequests/podcertificate_integration_test.go
@@ -497,7 +497,8 @@ func TestNodeAuthorization(t *testing.T) {
 	err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
 		_, err = node1Client.CertificatesV1beta1().PodCertificateRequests("default").Create(ctx, pcr, metav1.CreateOptions{})
 		if err != nil {
-			return false, err
+			t.Logf("Create PodCertificateRequest default/pcr1 as node1 failed with error: %v, retrying...", err)
+			return false, nil
 		}
 		return true, nil
 	})
@@ -506,7 +507,16 @@ func TestNodeAuthorization(t *testing.T) {
 	}
 
 	t.Run("node1 can directly get pcr1", func(t *testing.T) {
-		_, err := node1Client.CertificatesV1beta1().PodCertificateRequests("default").Get(ctx, "pcr1", metav1.GetOptions{})
+		// Getting the PCR could fail if there is lag in the node authorizer graph population.
+		// Poll until it succeeds.
+		err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+			_, err = node1Client.CertificatesV1beta1().PodCertificateRequests("default").Get(ctx, "pcr1", metav1.GetOptions{})
+			if err != nil {
+				t.Logf("Get PodCertificateRequest default/pcr1 as node1 failed with error: %v, retrying...", err)
+				return false, nil
+			}
+			return true, nil
+		})
 		if err != nil {
 			t.Fatalf("Unexpected error listing PodCertificateRequests as node1: %v", err)
 		}

--- a/test/integration/podcertificaterequests/podcertificate_integration_test.go
+++ b/test/integration/podcertificaterequests/podcertificate_integration_test.go
@@ -509,7 +509,7 @@ func TestNodeAuthorization(t *testing.T) {
 	t.Run("node1 can directly get pcr1", func(t *testing.T) {
 		// Getting the PCR could fail if there is lag in the node authorizer graph population.
 		// Poll until it succeeds.
-		err = wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
+		err := wait.PollUntilContextTimeout(ctx, 1*time.Second, 15*time.Second, true, func(ctx context.Context) (bool, error) {
 			_, err := node1Client.CertificatesV1beta1().PodCertificateRequests("default").Get(ctx, "pcr1", metav1.GetOptions{})
 			if err != nil {
 				t.Logf("Get PodCertificateRequest default/pcr1 as node1 failed with error: %v, retrying...", err)

